pyopensprinkler/station.py

```diff
@@ -1,42 +1,76 @@
 """Station module handling /station/ API calls."""
 
+import math
+
 
 class Station(object):
     """Station class with /station/ API calls."""
 
-    def __init__(self, opensprinkler, station, index):
+    def __init__(self, controller, index):
         """Station class initializer."""
-        self._opensprinkler = opensprinkler
-        self._station = station
+        self._controller = controller
         self._index = index
 
+    def _get_status_variable(self, statusIndex):
+        """
+        Retrieve station variable
+        
+        Program status data: each element is a 3-field array that stores the [pid,rem,start] of a station, where
+        pid is the program index (0 means none), rem is the remaining water time (in seconds), start is the start time.
+        If a station is not running (sbit is 0) but has a non-zero pid, that means the station is in the queue waiting to run.
+        """
+        return self._controller._state["settings"]["ps"][self._index][statusIndex]
+
+    def _manual_run(self, params=None):
+        """Manual station run"""
+        if params is None:
+            params = {}
+        params["sid"] = self._index
+        (_, content) = self._controller.request("/cm", params)
+        return content["result"]
+
+    def _bit_check(self, bit_property):
+        # [254, 255, 255]
+        # 254 = all but first station in the first block of 8 have master1 enabled
+        # each entry is for next block of 8 stations
+        bits = self._controller._state["stations"][bit_property]
+        bank = math.floor(self._index / 8)
+        bits = list(reversed([int(x) for x in list('{0:08b}'.format(bits[bank]))]))
+        position = (self._index % 8)
+
+        return bool(bits[position])
+
+    def run(self, seconds=60):
+        """Run station"""
+        params = {"en": 1, "t": seconds}
+        return self._manual_run(params)
+
+    def stop(self):
+        """Stop station"""
+        params = {"en": 0}
+        return self._manual_run(params)
+
+    def toggle(self):
+        """Toggle station"""
+        if self.is_running:
```

Comment:
I think there might be an issue here if the station is triggered from elsewhere or the run is finished as `is_running` maybe not have the right state. May have to do a `refresh()` here

Reply:
You mean do a `refresh()` before invoking just to ensure non-stale data?

Reply:
yup

Reply:
OK, I'll add it.

Reply:
Comitted.